src/classes/LVL_LevelAssign_BATCH.cls

```diff
@@ -270,32 +287,106 @@ public without sharing class LVL_LevelAssign_BATCH implements Database.Batchable
+        * @description Constructor
+        */
+        public Level(Level__c levelRecord) {
+            this.levelRecord = levelRecord;
+            targetIsAccount = target == String.valueOf(Account.SObjectType);
+            targetIsContact = target == String.valueOf(Contact.SObjectType);
+        }
+
+        /*********************************************************************************************************
+        * @description Builds query to get object data need to calculate and assign levels
+        * @returns SOQL string for querying SObjects for level calculation
+        */
+        public String buildQuery() {
+            String soqlQuery = 'SELECT Id';
+            for (String fieldName : getQueryFields()) {
+                soqlQuery += ', ' + fieldName;
             }
-            setFlds.add(lvl.Source_Field__c);
-            setFlds.add(lvl.Level_Field__c);
-            if (lvl.Previous_Level_Field__c != null)
-                setFlds.add(lvl.Previous_Level_Field__c);            
+            soqlQuery += ' FROM ' + target + buildFilter();
+            return soqlQuery;
+        }
+
+        /*********************************************************************************************************
+        * @description Builds set of fields needed for level query
+        * @returns Set of strings comprised of field names
+        */
+        private Set<String> getQueryFields() {
+            Set<String> queryFields = new Set<String>();
+            queryFields.add(sourceField);
+            queryFields.add(levelRecord.Level_Field__c);
+            if (!String.isEmpty(levelRecord.Previous_Level_Field__c)) {
+                queryFields.add(levelRecord.Previous_Level_Field__c);
+            }
+            return queryFields;
         }
         
-        return mapStrObjectToQueryFields; 
+        /*********************************************************************************************************
+        * @description Builds query to calulate level for the target object with filter if necessary
+        * @returns SOQL string
+        */
+        private String buildFilter() {
+            if (!shouldBuildFilter()) {
+                return '';
+            }
+
+            DateTime datetimeToFilter;
+            if (targetIsAccount) {
+                datetimeToFilter = levelsSettings.LastJobStartTimeAccount__c;
+            } else if (targetIsContact) {
+                datetimeToFilter = levelsSettings.LastJobStartTimeContact__c;
+            } else {
+                return '';
+            }
+
+            return ' WHERE SystemModStamp > ' + datetimeToFilter.formatGMT(DATETIME_SOQL_FORMAT);
+        }
+
+        /*********************************************************************************************************
+        * @description Determines is SystemModStamp filter should be built
```

Comment:
Change `is` to `if` and maybe add more information as to the logic used for the determination